pip/pip-433.md

```diff
@@ -0,0 +1,172 @@
+  - `defaultNumPartitions`: `2`
+- The namespace `public/default` exists, but you have not enabled Geo-Replication for the namespace yet.
+- Start a producer for a topic `persistent://public/default/topic` on the source cluster.
+  - It triggers a partitioned topic with `2` partitions created.
+- Enable namespace-level Geo-Replication for the namespace `public/default`.
+- Start geo replicator for the existing sub-topic `public/default/topic-partition-0`
+  - Without [PIP-414: Enforce topic consistency check](https://github.com/apache/pulsar/pull/24213), the geo replicator will trigger a non-partitioned topic creation, which is named `public/default/topic-partition-0`
+  - With [PIP-414: Enforce topic consistency check](https://github.com/apache/pulsar/pull/24213), the geo replicator get a denied error.
+- However, the user wants to allow the replicator to copy messages to the remote cluster.
+
+#### Issue 2: Replication is stuck because the remote side does not allow schema updates
+**Steps to reproduce the issue**
+- The topic`public/default/topic` has enabled a geo replication.
+- Users controls the topic schema manually, do not allow auto update schema by consumers/producers for both clusters.
+- The internal producer is stuck after the user sets a new schema on the source cluster.
+- However, the user wants to allowthe  replicator to copy the schema to the remote cluster.
+
+# Goals
+
+1. Add an optional choice: always allow the replicator to register schemas if compatible, even if users set `set-is-allow-auto-update-schema` to false.
+2. Add an optional choice: always allow the replicator to trigger a topic auto-creation, including partitioned topics, even if users have disabled topic auto-creation.
+  a. It only triggers auto-creation if the remote side has the same auto-creation policy of partition-type and partition-num.
+3. Checks compatibility between two clusters when enabling namespace-level replication, which includes the following
+  a. All topics’ partitions that have been created should be the same, including `__change_events`
+  b. Auto-creation policies should be the same
+  c. Support for adjusting policies automatically when enabling namespace-level policies, and provide an API to review the changes that will be applied. By the way, this feature will be an option param of enabling namespace-level geo replication
+
+# Detailed Design
+
+### Public API
+
+#### Regarding Goal 1: "always allow the replicator to register schemas if compatible"
+
+**The original design of `pulsar-admin topics set-is-allow-auto-update-schema`**
+
+```shell
+pulsar-admin namespaces set-is-allow-auto-update-schema
+  -d, --disable         Disable schema validation enforced
+  -e, --enable          Enable schema validation enforced
+```
+
+To add a new param `--always-allow-for-replicator-if-compatible`, which means that always allow the replicator to register a new schema if compatible. The default value is `true`.
```

Comment:
Do we need this option in broker.conf?

Reply:
After this PIP, the broker-level configuration `allowAutoTopicCreation` will never affect the behavior of the replicator anymore.